kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt

```diff
@@ -1,72 +1,1061 @@
+            }
+          }
+          
+          "Positive Infinity" {
+            onDoubleRange { double, _ ->
+              double shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+          }
+          
+          "NaN" {
+            onDoubleRange { double, _ ->
+              double shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+        
+      }
+      
+      "The non-numeric double" - {
+        "NaN" - {
+          "Should not be greater than or equal to" - {
+            "Itself" {
+              NaN shouldNotMatchGreaterThanOrEqual NaN
+            }
+            
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                NaN shouldNotMatchGreaterThanOrEqual it
+              }
+            }
+            
+            "Positive Infinity" {
+              NaN shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "Negative Infinity" {
+              NaN shouldNotMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+          }
+        }
+        
+        "Positive Infinity" - {
+          "Should be greater than or equal to" - {
+            "Itself" {
+              POSITIVE_INFINITY shouldMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "Negative Infinity" {
+              POSITIVE_INFINITY shouldMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+            
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                POSITIVE_INFINITY shouldMatchGreaterThanOrEqual it
+              }
+            }
+          }
+          
+          "Should not be greater than or equal to" - {
+            "NaN" {
+              POSITIVE_INFINITY shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+        
+        "Negative Infinity" - {
+          "Should be greater than or equal to" - {
+            "Itself" {
+              NEGATIVE_INFINITY shouldMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+          }
+          
+          "Should not be greater than or equal to" - {
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual it
+              }
+            }
+            
+            "Positive Infinity" {
+              NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "NaN" {
+              NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+      }
+    }
+    
+    
+  }
+  
+  private fun shouldThrowAssertionError(message: String, vararg expression: () -> Any?) {
+    expression.forEach {
+      val exception = shouldThrow<AssertionError>(it)
+      exception.message shouldBe message
+    }
+  }
+  
+  private infix fun Double.shouldExactlyMatch(other: Double) {
```

Comment:
I like these functions that increase coverage without duplication 👍

Reply:
They're good because we expect the exact same behavior now. But this is dangerous, because if I change any behavior in the future, the old tests wouldn't cover them. This way we can cover on all fronts, including trying the exception ones